src/main/java/io/strimzi/kafka/bridge/http/HttpSourceBridgeEndpoint.java

```diff
@@ -66,6 +66,12 @@ public void open() {
         super.open();
     }
 
+    public void closeOnError() {
```

Comment:
why this name? I mean, from the code inside, it doesn't show up that it's closing when there is an error.

Reply:
Do you have a suggestion for the name? I'm just trying to keep the `Cyclomatic Complexity` within the bounds without changing too much code with this PR.

Reply:
Let's try with something like `maybeClose` taking into account that the `closing` has to be validate.